crates/sui/src/client_commands.rs

```diff
@@ -2901,3 +2903,29 @@ pub(crate) async fn prerender_clever_errors(
         }
     }
 }
+
+/// Execute a transaction and wait for the effects cert.
+/// The transaction execution is not guaranteed to succeed and may fail. This is usually only
+/// needed in non-test environment or the caller is explicitly testing some failure behavior.
+///
+/// Used only in the CLI as it needs to wait for the effects cert instead of local execution.
+/// Use the `execute_transaction_may_fail` from WalletContext
+/// for most cases where read-after write consistency is needed.
+pub async fn execute_tx(
```

Comment:
I think it's useful to add this function to `SuiClient`

Reply:
You mean in the SDK?

Reply:
Moved it to SuiClient and updated the `client_commands.rs` to use it!